src/analysis/retail/druid/balance/modules/spells/GatheringStarstuff.tsx

```diff
@@ -0,0 +1,263 @@
+  stacksWhenCast: number[][];
+  alreadyMaxStacksWhenCast = 0;
+  gastBuffedAbilities: { [key: number]: number } = {};
+  gastActivationTimestamp: number | null = null;
+  gastConsumptionTimestamp: number | null = null;
+  gastCurrentStacks = 0;
+  totalStacks = 0;
+  //Wrath
+  listCastWrath: number[];
+  listCastWrathTimestamps: number[];
+  //Starfire
+  listCastStarfire: number[];
+  listCastStarfireTimestamps: number[];
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.has2PieceByTier(TIERS.T29);
+    this.buffStacks = Array.from({ length: GATHERING_STARSTUFF.MAX_STACKS + 1 }, (x) => [0]);
+    this.stacksWhenCast = Array.from({ length: GATHERING_STARSTUFF.MAX_STACKS + 1 }, (x) => [0]);
+    this.listCastWrath = [];
+    this.listCastWrathTimestamps = [];
+    this.listCastStarfire = [];
+    this.listCastStarfireTimestamps = [];
+
+    Object.values(GATHERING_STARSTUFF.AFFECTED_CAST).forEach((spell) => {
+      this.gastBuffedAbilities[spell.id] = 0;
+    });
+
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(GATHERING_STARSTUFF.AFFECTED_CAST),
+      this.onCast,
+    );
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER).spell(GATHERING_STARSTUFF.AFFECTED_DAMAGE),
+      this.onDamage,
+    );
+    this.addEventListener(
+      Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.CONVOKE_SPIRITS),
+      this.onEndConvoke,
+    );
+
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.GATHERING_STARSTUFF),
+      this.handleStacks,
+    );
+    this.addEventListener(
+      Events.applybuffstack.by(SELECTED_PLAYER).spell(SPELLS.GATHERING_STARSTUFF),
+      this.handleStacks,
+    );
+    this.addEventListener(
+      Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.GATHERING_STARSTUFF),
+      this.handleStacks,
+    );
+    this.addEventListener(
+      Events.refreshbuff.by(SELECTED_PLAYER).spell(SPELLS.GATHERING_STARSTUFF),
+      this.handleStacks,
+    );
+    this.addEventListener(Events.fightend, this.handleStacks);
+  }
+
+  // As stacks are consumed when the cast end but the damage are done when the spell lands, we need
+  // to match CastEvent (on which we obtain the number of stacks) with their respective DamageEvent.
+  // To do so we list all CastEvent and when a DamageEvent is read we check with the first CastEvent
+  // on the list if they are in an acceptable time frame. If so we know how many stacks are associated
+  // with a DamageEvent. We can then discard the first CastEvent and go on with our Event. If not,
+  // we either discard the damage event if it's to early (can happen with precast or convoke
+  // cast) or the cast event if it's to late (can happen with a targt dying after the cast but
+  // before the spell lands).
+
+  onCast(event: CastEvent) {
+    const stacks = this.selectedCombatant.getBuff(SPELLS.GATHERING_STARSTUFF.id)?.stacks;
+    this.gastCurrentStacks = stacks ?? 0;
```

Comment:
remove this class variable and just use stacks. You don't seem to use it anywhere else in the class.